src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -379,89 +388,215 @@ internal Database(C4Database* c4db)
+        /// <exception cref = "CouchbaseLiteException" > Thrown with <see cref="C4ErrorCode.NotFound"/>
+        /// if <see cref="Database"/> is closed</exception>
         public Scope GetDefaultScope()
         {
-            throw new NotImplementedException();
+            ThreadSafety.DoLocked(() =>
+            {
+                CheckOpen();
+                if (_defaultScope == null) {
+                    _defaultScope = new Scope(this);
+                    if(!_scopes.ContainsKey(_defaultScopeName))
+                        _scopes.TryAdd(_defaultScopeName, _defaultScope);
+        }
+            });
+
+            return _defaultScope;
         }
 
         /// <summary>
         /// Get the default collection.
         /// </summary>
-        /// <returns></returns>
+        /// <remarks>
+        /// None recoverable default collection can be deleted from the database. Once default collection is deleted from db, 
+        /// it will be gone for good.
+        /// </remarks>
+        /// <returns>default collection</returns>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        /// <exception cref = "CouchbaseLiteException" > Thrown with <see cref="C4ErrorCode.NotFound"/>
+        /// if <see cref="Database"/> is closed</exception>
         public Collection GetDefaultCollection()
         {
-            throw new NotImplementedException();
+            ThreadSafety.DoLocked(() =>
+            {
+                CheckOpen();
+                if (_defaultCollection == null) {
+                    var c4coll = (C4Collection*)LiteCoreBridge.Check(err =>
+                    {
+                        return Native.c4db_getDefaultCollection(c4db, err);
+                    });
+
+                    if (c4coll != null) {
+                        _defaultCollection = new Collection(this, _defaultCollectionName, GetDefaultScope(), c4coll);
+                    }
+                }
+            });
+
+            return _defaultCollection;
         }
 
         /// <summary>
         /// Get scope names that have at least one collection.
-        /// Note: the default scope is exceptional as it will always be listed even though there are no collections    
-        /// under it.
+        /// The default scope is an exception as it will always be listed even it doesn't contain any collection.    
         /// </summary>
-        /// <returns></returns>
+        /// <returns>
+        /// scope names of all existing scopes, in the order in which they were created.
+        /// </returns>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        /// <exception cref = "CouchbaseLiteException" > Thrown with <see cref="C4ErrorCode.NotFound"/>
+        /// if <see cref="Database"/> is closed</exception>
         public IReadOnlyList<Scope> GetScopes()
```

Comment:
What is the difference between this and the Scopes property?

Reply:
Let me remove the Scope property since it's not defined in the API doc and it's not that useful to keep it around as internal either.

Reply:
Something similar going on in the scope class (Collections / GetCollections)

Reply:
That Collections property under Scope I rather keep because I can get the cached collection list internally much faster than using GetCollections method.